src/app/services/data.service.ts

```diff
@@ -0,0 +1,172 @@
+      date: '11.03.2019',
+      time: '14:03:01',
+      author: 'no name',
+      source: 'CNN'
+    },
+    {
+      id: '9',
+      title: 'Text 9',
+      shortDescription: 'Lorem ipsum dolor sit amet, consectetur adipisicing',
+      content: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iusto animi architecto voluptatum sapiente natus ipsam eos, totam magnam corrupti facilis?',
+      date: '11.02.2019',
+      time: '10:01:01',
+      author: 'Vitali',
+      source: 'The inthernet'
+    },
+    {
+      id: '10',
+      title: 'Text 10',
+      shortDescription: 'Lorem ipsum dolor sit amet, consectetur adipisicing',
+      content: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iusto animi architecto voluptatum sapiente natus ipsam eos, totam magnam corrupti facilis?',
+      date: '11.03.2019',
+      time: '14:04:01',
+      author: 'John',
+      source: 'BBC'
+    }
+  ];
+
+  public newsSource = new BehaviorSubject<object>(this.newsItems.slice(0, 5));
+  public currentNews = this.newsSource.asObservable();
+
+  constructor(private router: Router) { }
+
+  public delete(listItems: News[] | undefined, id: string): void {
+      const deleteNews = listItems.find((item: News) => item.id === id);
+      listItems.splice(listItems.indexOf(deleteNews), 1);
+      this.newsSource.next(listItems);
+      this.router.navigate(['/main']);
+  }
+
+  public loadNews(currentItems: News[]): void {
+    this.newsSource.next(currentItems.concat(this.newsItems.slice(5)));
+  }
+
+  public getItem(id: string): News {
+    return this.newsItems.find((item: News) => item.id === id);
+  }
+
+  public filterByMe(): void {
+    this.newsSource.next(this.newsItems.filter((item: News) => item.author === 'Vitali'));
+  }
+
+  public getAllItems(): void {
+    this.newsSource.next(this.newsItems.slice(0, 5));
+  }
+
+  public filterBySource(value: string): void {
+    this.newsSource.next(this.newsItems.filter((item: News) => item.source === value));
+  }
+
+  public filterByKeyWords(value: string | undefined,
+                          source: string | undefined,
+                          author: string | undefined,
+                          data: News[] | undefined): void {
+    let newNews: News[];
+
+    if (value && author) {
+      newNews = data.filter((item: News) => (item.content.indexOf(value) !== -1) && (item.author === author));
+
+    } else if (value && source) {
+      newNews = data.filter((item: News) => (item.content.indexOf(value) !== -1) && (item.source === source));
+
+    } else if (value && !source) {
+      newNews = data.filter((item: News) => (item.content.indexOf(value) !== -1));
+
+    } else if (!value && source) {
+      newNews = data.filter((item: News) => item.source === source);
+
+    } else if (!value && !source && !author) {
+      newNews = this.newsItems.slice(0, 5);
+    }
```

Comment:
if we want to add another filter? should we change the code here and in other components? what if to create method tht is more flexible